rest-api/src/main/java/com/cloudbees/workflow/rest/external/FlowNodeLogExt.java

```diff
@@ -119,7 +119,14 @@ public static FlowNodeLogExt create(FlowNode node) {
                 if (logLen > 0) {
                     StringWriter writer = new StringWriter();
                     try {
-                        logText.writeHtmlTo(logLen - logExt.getLength(), writer);
+                        long pos = 0;
+                        while (true) {
+                            long pos2 = logText.writeHtmlTo(logLen - logExt.getLength(), writer);
+                            if (pos2 <= pos) {
+                                break;
+                            }
+                            pos = pos2;
+                        }
```

Comment:
Old code gives a spotbugs warning against new versions of `workflow-api`. Would not matter unless there was bunch of log text for a single step, but the fix is to just copy https://github.com/jenkinsci/workflow-job-plugin/blob/e94f51a47011510948bf9bdc2f6e94efa0768055/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L1109.